.github/workflows/backend.yml

```
@@ -86,37 +62,56 @@ jobs:
           client/python/requirements.txt
           requirements.txt
           test/requirements.txt
-    - name: Install Client
+    - name: Create env
       run: |
-        pip install -e .
+        python -m pip install --upgrade virtualenv
+        python -m virtualenv venv
+    - uses: actions/cache@master
+      id: cache
+      with:
+        path: |
+          client/python/venv/*
+        key: python-client-${{ runner.os }}-pip-${{ hashFiles('client/python/requirements.txt') }}-${{ hashFiles('client/python/test/requirements.txt') }}
+        restore-keys: |
+          python-client-${{ runner.os }}-pip-
+          python-client-${{ runner.os }}-
+    - name: Install Test Requirements (Linux)
+      if: steps.cache.outputs.cache-hit != 'true' && runner.os == 'Linux'
+      run: |
+        . venv/bin/activate
+        python -m pip install -r test/requirements.txt
+    - name: Install Client Library (Linux)
+      if: runner.os == 'Linux'
```

This does not need to be run if the cache is found, right? 
```suggestion
      if: steps.cache.outputs.cache-hit != 'true' and runner.os == 'Linux'
```